src/Tasks/AssemblyDependency/GlobalAssemblyCache.cs

```diff
@@ -367,16 +367,16 @@ bool specificVersion
         }
 
         /// <summary>
-        /// Return the root path of the GAC
+        /// Return the root path of the GAC.
         /// </summary>
         internal static string GetGacPath()
         {
             int gacPathLength = 0;
             NativeMethods.GetCachePath(AssemblyCacheFlags.GAC, null, ref gacPathLength);
-            StringBuilder gacPath = new StringBuilder(gacPathLength);
+            char[] gacPath = new char[gacPathLength];
```

Comment:
Stackalloc for all of these. Generally you should use it where the buffer is not returned, is used synchronously, and you need no more then about 512 bytes as a rule of thumb. Zero allocation